internal/generate/pg/model.go.tmpl

```diff
@@ -114,9 +118,9 @@ SELECT
 {{- end }}
 FROM {{ .Table.SchemaName }}.{{ .Table.TableName }}
 WHERE TRUE
-{{- range $selectFields }}
-  AND (CAST(:{{ .Column.ColumnName }} AS {{ .Column.Type | ToUpper }}) IS NULL or {{ .Column.ColumnName }} = :{{ .Column.ColumnName }})
-{{- end }}
+{{- range $pkFields }}
```

Comment:
only `pkFields` not necessarily be the case. 
1. what about the scenario where columns one or more columns have a unique constraint but are not explicitly annotated as primary key?
2. what if we've multiple id columns, one explicitly annotated as primary key and another unique auto incrementing column defined by an arbitrary use case? what if i want to find or delete by this business id key?

hence `IS NULL` check takes care of a use-case wherein we don't have to provide all the fields

Reply:
If there are other ways to select a single row, the FindAll should suffice as it already has the ability to build the Where with any provided fields. It's not too difficult to pull row[0] from that result.

Reply:
That said, there is currently a bug in the find query, including a wayward semi-colon before LIMIT. I'll have another PR on it soon.  :)

Templates for queries have not been updated to match - yet.

Reply:
> If there are other ways to select a single row, the FindAll should suffice as it already has the ability to build the Where with any provided fields. It's not too difficult to pull row[0] from that result.

that was my thought initially, but from the perspective of someone who is consuming this generated code, checking for both `error` and then `length` and `first element` check gets repetitive and tedious very soon which led me to the idea of baking that in!

so this way, you've the option to do both and doesn't make assumptions on how this is consumed. if that makes any sense!